build/images/base/build.sh

```diff
@@ -129,29 +135,73 @@ if $PULL; then
-       --build-arg CNI_BINARIES_VERSION=$CNI_BINARIES_VERSION \
-       --build-arg BUILD_TAG=$BUILD_TAG .
-
-if [ "$DISTRO" == "ubuntu" ]; then
-    docker build $PLATFORM_ARG \
-           --cache-from antrea/cni-binaries:$CNI_BINARIES_VERSION \
-           --cache-from antrea/base-ubuntu:$BUILD_TAG \
-           -t antrea/base-ubuntu:$BUILD_TAG \
-           --build-arg CNI_BINARIES_VERSION=$CNI_BINARIES_VERSION \
-           --build-arg SURICATA_VERSION=$SURICATA_VERSION \
-           --build-arg BUILD_TAG=$BUILD_TAG .
-elif [ "$DISTRO" == "ubi" ]; then
-    docker build $PLATFORM_ARG \
-           --cache-from antrea/cni-binaries:$CNI_BINARIES_VERSION \
-           --cache-from antrea/base-ubi:$BUILD_TAG \
-           -t antrea/base-ubi:$BUILD_TAG \
-           -f Dockerfile.ubi \
-           --build-arg CNI_BINARIES_VERSION=$CNI_BINARIES_VERSION \
-           --build-arg SURICATA_VERSION=$SURICATA_VERSION \
-           --build-arg BUILD_TAG=$BUILD_TAG .
+if [ "$TYPE" == "unified" ]; then
+    docker build $PLATFORM_ARG --target cni-binaries \
+        --cache-from antrea/cni-binaries:$CNI_BINARIES_VERSION \
+        -t antrea/cni-binaries:$CNI_BINARIES_VERSION \
+        --build-arg CNI_BINARIES_VERSION=$CNI_BINARIES_VERSION \
+        --build-arg BUILD_TAG=$BUILD_TAG .
+
+    if [ "$DISTRO" == "ubuntu" ]; then
+        docker build $PLATFORM_ARG \
+            --cache-from antrea/cni-binaries:$CNI_BINARIES_VERSION \
+            --cache-from antrea/base-ubuntu:$BUILD_TAG \
+            -t antrea/unified-base-ubuntu:$BUILD_TAG \
+            --build-arg CNI_BINARIES_VERSION=$CNI_BINARIES_VERSION \
+            --build-arg SURICATA_VERSION=$SURICATA_VERSION \
+            --build-arg BUILD_TAG=$BUILD_TAG .
+    elif [ "$DISTRO" == "ubi" ]; then
+        docker build $PLATFORM_ARG \
+            --cache-from antrea/cni-binaries:$CNI_BINARIES_VERSION \
+            --cache-from antrea/base-ubi:$BUILD_TAG \
+            -t antrea/base-ubi:$BUILD_TAG \
+            -f Dockerfile.ubi \
+            --build-arg CNI_BINARIES_VERSION=$CNI_BINARIES_VERSION \
+            --build-arg SURICATA_VERSION=$SURICATA_VERSION \
+            --build-arg BUILD_TAG=$BUILD_TAG .
+    fi
+elif [ "$TYPE" == "agent" ]; then
+    docker build $PLATFORM_ARG --target cni-binaries \
+        --cache-from antrea/cni-binaries:$CNI_BINARIES_VERSION \
+        -t antrea/cni-binaries:$CNI_BINARIES_VERSION \
+        --build-arg CNI_BINARIES_VERSION=$CNI_BINARIES_VERSION \
+        --build-arg BUILD_TAG=$BUILD_TAG .
+
+    if [ "$DISTRO" == "ubuntu" ]; then
+        docker build $PLATFORM_ARG \
+            --cache-from antrea/cni-binaries:$CNI_BINARIES_VERSION \
+            --cache-from antrea/base-ubuntu:$BUILD_TAG \
+            -t antrea/agent-base-ubuntu:$BUILD_TAG \
+            --build-arg CNI_BINARIES_VERSION=$CNI_BINARIES_VERSION \
+            --build-arg SURICATA_VERSION=$SURICATA_VERSION \
+            --build-arg BUILD_TAG=$BUILD_TAG .
```

Comment:
same, here only difference is "agent" instead of "unified" while tagging. Can use $TYPE variable and refactor it to have these lines only once ?

Reply:
yes the refactoring part is left to be done, its just a draft.